### ISSUE TITLE
Global config option to disable parsing and creation of anything out of a cache listing's personal cache notes

### ISSUE THREAD

**Issue by baiti** [cgeo/cgeo#8848]
Quite a few geocaching applications have started to create waypoints based on coordinates that they have extracted from a cache listing's personal cache notes (PCN). c'geo does that too. PCN is a free form scratch pad for users to take notes for geocaches. There is no defined format of how PCNs are structured and consequently there is no defined way for a program to parse any meaningful information out of the PCN. Some applilcations, for instance project-gc provides a way for the user to mask coordinates in the PCN with a prefixed caret e.g. "^N48 44.444 E08 44.444" and thus prevent project-gc from creating a waypoint (and render it on the map) for coordinates masked in this way. 

This I'd consider a circumvention of a problem but not a true solution, however, it is still better than nothing and if you for any reason cannot implement the functionality I am going to describe below, at least provide the masking capability by parsing for the caret "^" prefix.

A solution would be a system wide configuration option that allows a user to instruct c'geo to completely avoid parsing the PCN data for anything.

I am aware that recent versions of c'geo provide a checkbox underneath the PCN Editor that reads "Prevent waypoint extraction". While I fully understand that checkboxes semantics, it addresses only a part of the problem as I have no way to tell c'geo to stay away from the contents of the PCN field entirely. c'geo's waypoint extraction function happens at the time a geocache listing is downloaded (or refreshed) from geocaching.com and persisted for offline use in cgeo.

**Comment by Lineflyer** [comment]
Fair enough.
Maybe it is possible to implement it in a way, that the global setting (once active) triggers the checkbox in the PN field to be active by default for all caches (so users can deactivate it for single caches, meaning kind of opt-in instead of opt-out).

**Comment by eddiemuc** [comment]
> Fair enough.
> Maybe it is possible to implement it in a way, that the global setting (once active) triggers the checkbox in the PN field to be active by default for all caches (so users can deactivate it for single caches, meaning kind of opt-in instead of opt-out).
> 

I support this idea and I think it should be easy to implement

**Comment by Lineflyer** [comment]
Go for it, if you would like to implement this.
I do however suggest to keep the current "opt-out" behavior as default so the behavior does not change compared to the current market version.

**Comment by baiti** [comment]
> Go for it, if you would like to implement this.
> I do however suggest to keep the current "opt-out" behavior as default so the behavior does not change compared to the current market version.

are you addressing this to me? I sure could give it a try, however, I am not familiar (yet) with the development process. Can I do local builds on my development system? How do I test? Where do I commit to? Is there a sandbox such that I don't contaminate production code? Can I make changes to other developper's modules? Is there any quick introduction into all this?

**Comment by baiti** [comment]
> > Go for it, if you would like to implement this.
> > I do however suggest to keep the current "opt-out" behavior as default so the behavior does not change compared to the current market version.
> 
> are you addressing this to me? I sure could give it a try, however, I am not familiar (yet) with the development process. Can I do local builds on my development system? How do I test? Where do I commit to? Is there a sandbox such that I don't contaminate production code? Can I make changes to other developper's modules? Is there any quick introduction into all this?

Found the Wiki ... that will get me going ;)

**Comment by Lineflyer** [comment]
@baiti 
I originally adressed it to @eddiemuc as it sounded like he wants to look into it.
BUT...you are more than welcome to contribute. We are currently a rather small team, so any help is welcome. I would be happy to see more contributors. 
In case of any questions to setup your environment, just ask the team here and someone will help you.

**Comment by eddiemuc** [comment]
Hi @baiti , as @Lineflyer already said, it would be wonderful if you see a chance to contribute to the project. If you have any troubles getting started don't hesitate to contact me or any other member of the team.
I think you already found the start page at https://github.com/cgeo/cgeo

**Comment by baiti** [comment]
> Hi @baiti , as @Lineflyer already said, it would be wonderful if you see a chance to contribute to the project. If you have any troubles getting started don't hesitate to contact me or any other member of the team.
> I think you already found the start page at https://github.com/cgeo/cgeo

Have installed Android Studio, SDKs, got  API keys (had those already), have accomplished a first build of an APK and have managed to get that running in the emulator. Quite impressive. I guess I could now start changing code. It will take me some time to work myself through some of the code. I have found the place were extractWaypoints is done and I could code a conditional around it to execute it only if the respective config option is set. Now it is about adding another entry to the settings dialog. It appears to me that a good place would be under:

Settings -> Offline Data, right below the entry "Additional lists"

So it would be:

Settings -> Offline Data -> Waypoint Extraction
Disable waypoint extration from Personacl Cache Notes

Would that make sense to you guys from an overall UX perspective?

**Comment by Lineflyer** [comment]
> Settings -> Offline Data -> Waypoint Extraction
> Disable waypoint extration from Personacl Cache Notes

I would rather expect it under Settings -> Cache Details as this page also contains settings related to what you get when loading a cache. Lets wait for other opinions.

**Comment by baiti** [comment]
> > Settings -> Offline Data -> Waypoint Extraction
> > Disable waypoint extration from Personacl Cache Notes
> 
> I would rather expect it under Settings -> Cache Details as this page also contains settings related to what you get when loading a cache. Lets wait for other opinions.

Yes that makes more sense indeed.

**Comment by eddiemuc** [comment]
@baiti 
I would also expect this under Settings -> Cache Data 

Regarding how to apply this to the cache:
There already exists an attribute `preventWaypointsFromNote` of class `Geocache`. If this is set to `true` no waypoints should be extracted from note. Problem is that it is set to `false` initially. So what I would try to do is set the initial value of this attribute simply to the respective new value from the ``Settings` that you want to introduce. Hopefully the logic for preventing the actual extraction should then already be there.
Does that makes sense?

**Comment by baiti** [comment]
> @baiti
> I would also expect this under Settings -> Cache Data
> 
> Regarding how to apply this to the cache:
> There already exists an attribute `preventWaypointsFromNote` of class `Geocache`. If this is set to `true` no waypoints should be extracted from note. Problem is that it is set to `false` initially. So what I would try to do is set the initial value of this attribute simply to the respective new value from the ``Settings` that you want to introduce. Hopefully the logic for preventing the actual extraction should then already be there.
> Does that makes sense?

I have it coded such that it works in the emulator. I can't do much on translation of the strings into different languages but before we get to this, I'd like some of you experts have a look at what I have done. How do we do this?

**Comment by bekuno** [comment]
> I have it coded such that it works in the emulator. I can't do much on translation of the strings into different languages but before we get to this, I'd like some of you experts have a look at what I have done. How do we do this?

see https://github.com/cgeo/cgeo/wiki/git-beginners

In AS: 
Tools -> Tasks & Kontexts -> Open Tasks -> define task & changelist & create branch from master
work at your branch
from changelist: Commit File ...
in the dialog: 
  - before commit: Performe code analysis
  - Commit and Push
at cgeo github: refresh the pull request page
you shoud see a "compare and pull request" entry on top
create the pull request; please check that only your commit is shown

**Comment by baiti** [comment]
> > I have it coded such that it works in the emulator. I can't do much on translation of the strings into different languages but before we get to this, I'd like some of you experts have a look at what I have done. How do we do this?
> 
> see https://github.com/cgeo/cgeo/wiki/git-beginners
> 
> In AS:
> Tools -> Tasks & Kontexts -> Open Tasks -> define task & changelist & create branch from master
> work at your branch
> from changelist: Commit File ...
> in the dialog:
> 
> * before commit: Performe code analysis
>
> * Commit and Push

up to this point it worked, it says 5 files commmitted.

>   at cgeo github: refresh the pull request page

push failed, permission denied

consequently I couldn't do these steps:
>   you shoud see a "compare and pull request" entry on top
>   create the pull request; please check that only your commit is shown

Do I need some authorization? Or was it some sort of a user error on my side?

**Comment by bekuno** [comment]
> push failed, permission denied

Hmm, 
you did nor fork the project.
See https://github.com/cgeo/cgeo#get-the-source

Then you have:
~~~
>git remote -v
origin  https://github.com/bekuno/cgeo.git (fetch)
origin  https://github.com/bekuno/cgeo.git (push)
upstream        https://github.com/cgeo/cgeo (fetch)
upstream        https://github.com/cgeo/cgeo (push)
~~~

If this is correct, then you push to your clone.
And the PR is typically between your special branch and cgeo master branch.

**Comment by baiti** [comment]
> > push failed, permission denied
> 
> Hmm,
> you did nor fork the project.

Exactly! When I set up the project in AS, I used the "git clone" link without creating a fork first .. and using the fork link then.
